osquery/TPipe.py

```diff
@@ -0,0 +1,204 @@
+        if len(buff) == 0:
+            raise TTransportException(
+                type=TTransportException.END_OF_FILE,
+                message='TPipe read 0 bytes')
+        return buff
+
+    def write(self, buff):
+        if not self.isOpen():
+            raise TTransportException(
+                type=TTransportException.NOT_OPEN,
+                message='Called read on non-open pipe')
+
+        bytesWritten = None
+        try:
+            (writeError, bytesWritten) = win32file.WriteFile(
+                self._handle, buff, None)
+        except Exception as e:
+            raise TTransportException(
+                type=TTransportException.UNKNOWN,
+                message='Failed to write to named pipe: ' + e.message)
+
+    def flush(self):
+        win32file.FlushFileBuffers(self._handle)
+
+
+class TPipeServer(TPipeBase, TServerTransportBase):
+    """Named pipe implementation of TServerTransport"""
+
+    def __init__(self, pipeName, buffsize=4096, maxconns=255):
+        """
+        Initialize a TPipe client
+
+        @param pipeName(string)  The path of the pipe to connect to.
+        @param buffsize(int)  The size of read/write buffers to use.
+        @param maxconns(int)  Maximum number of simultaneous connections.
+        """
+        self._pipeName = pipeName
+        self._buffsize = buffsize
+        self._maxconns = maxconns
+        self._handle = None
+
+        # Overlapped event for Windows Async IO
+        self._overlapped = pywintypes.OVERLAPPED()
+        self._overlapped.hEvent = win32event.CreateEvent(None, 0, 0, None)
+
+    # Create a new named pipe if one doesn't already exist
+    def listen(self):
+        if self._handle == None:
+            ret = self.createNamedPipe()
+
+        if ret != True:
+            raise TTransportException(
+                type=TTransportException.NOT_OPEN,
+                message='TCreateNamedPipe failed')
+
+    def accept(self):
+        if self._handle == None:
+            self.createNamedPipe()
+
+        self.initiateNamedConnect()
+        result = TPipe(self._pipeName)
+        result.setHandle(self._handle)
+
+        return result
+
+    def createNamedPipe(self):
+        openMode = win32pipe.PIPE_ACCESS_DUPLEX | win32file.FILE_FLAG_OVERLAPPED
+        pipeMode = win32pipe.PIPE_TYPE_BYTE | win32pipe.PIPE_READMODE_BYTE
+
+        saAttr = pywintypes.SECURITY_ATTRIBUTES()
+        saAttr.SetSecurityDescriptorDacl(1, None, 0)
+
+        self._handle = win32pipe.CreateNamedPipe(
+            self._pipeName, openMode, pipeMode,
+            win32pipe.PIPE_UNLIMITED_INSTANCES, self._buffsize, self._buffsize,
+            win32pipe.NMPWAIT_WAIT_FOREVER, saAttr)
+
+        err = win32api.GetLastError()
+        if self._handle.handle == winerror.ERROR_INVALID_HANDLE:
+            raise TTransportException(
+                type=TTransportException.NOT_OPEN,
+                message='TCreateNamedPipe failed: {}'.format(err))
+            return False
```

Comment:
I think this is unreachable as you're raising an exception before. Also for consistency maybe it makes sense to throw on error and not return as you're doing on other methods.